src/teehr/visualization/dataframe_accessor.py

```diff
@@ -28,15 +32,60 @@ class TEEHRDataFrameAccessor:
     def __init__(self, pandas_obj):
         """Initialize the class."""
         self._df = pandas_obj
+        self._df.attrs = pandas_obj.attrs
         self._validate(pandas_obj)
 
     @staticmethod
     def _validate(obj):
-        """Validate the DataFrame object columns."""
-        if "location_id" not in obj.columns:
-            raise AttributeError("Must have 'location_id'.")
-        if obj.index.size == 0:
-            raise AttributeError("DataFrame must have data.")
+        """Validate the DataFrame object."""
+        if 'table_type' not in obj.attrs:
+            raise AttributeError("Not DataFrame Attribute 'table_type'"
+                                 " defined.")
```

Comment:
I would tend to:
```
raise AttributeError(
    "Not DataFrame Attribute 'table_type'" defined."
)
```
instead of splitting strings across lines.

Reply:
Attempted to reformat all strings to keep on minimal lines and adhere to Flake08 guidelines. Let me know if you had something else in mind or the indentation looks off.